static/vendor/javascripts/prebid/safeframe.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+
+// this script can be returned by an ad server delivering a cross domain iframe, into which the
+// creative will be rendered, e.g. DFP delivering a SafeFrame
+
+// It is strongly inspired by Prebid.js/integrationExamples/gpt/x-domain/creative.js
+
+// Use this in combination with a creative modelled on:
+// https://www.google.com/dfp/59666047#delivery/PreviewCreative/orderId=2259406532&lineItemId=4615205472&creativeId=138228201638
+//
+
+var guardian_pb_safeframe = {
+    "adServerDomain": function(){
+        return window.location.protocol + '//tpc.googlesyndication.com';
+    },
+    "renderAd": function(ev) {
+        var key = ev.message ? 'message' : 'data';
+        var adObject = {};
+        try {
+            adObject = JSON.parse(ev[key]);
+        } catch (e) {
+            return;
```

Comment:
Does it matter if the parsing failed?  ie. should this be logged or is it OK to ignore it?

Reply:
I think failure is expected, as a way to filter out messages from other channels.